pkg/rhcos/qemu.go

```diff
@@ -24,5 +25,22 @@ func QEMU(ctx context.Context) (string, error) {
 		return "", err
 	}
 
-	return base.ResolveReference(relQEMU).String(), nil
+	baseURL := base.ResolveReference(relQEMU).String()
+
+	// Attach sha256 checksum to the URL. If the file has the ".gz" extension, then the
+	// data is compressed and we use SHA256 value; otherwise we work with uncompressed
+	// data and therefore need UncompressedSHA256.
+	if strings.HasSuffix(baseURL, ".gz") {
```

Comment:
if we are doing to decompress always, maybe we only care about the uncompressed one. why do both?

Reply:
Good point.

Reply:
It's definitely incorrect.  Using the uncompressed checksum here results in the client attempting to checksum the received (compressed) file against the uncompressed checksum, which of course fails.

Reply:
I was more asking do you care about the checksum of the downloaded file if we always decompress? we need to make sure the contents are correct in transport, that means we could download->uncompress->then check for cehcksum.

Reply:
I suppose, although it would mean changing the code there => risk of introducing new bugs.

Reply:
> https://play.golang.org/p/kdJwsENHuUH

that allows the cache to uncompress always un-fly without more tmp files, and then check hash of uncompressed data.

Reply:
also @RobertKrawitz if ^^ simplifies some of the logic..

Reply:
There were two ways I could go about doing it (either keep the pkg/rhcos stuff as is and checksum through a pipe prior to decompression, or change the pkg/rhcos stuff to only provide the uncompressed checksum).  I went back and forth on it a few times but eventually decided to bite the bullet and always provide the uncompressed checksum.

Since the filename is based on the checksum of the URL, any old filenames in the cache will be invalidated (verified) and I could remove that code (which will be dead now, modulo hash collisions!)

Reply:
> Since the filename is based on the checksum of the URL, any old filenames in the cache will be invalidated (verified) and I could remove that code (which will be dead now, modulo hash collisions!)

yeah, would like to drop the `getCachedFile` changes, we shouldn't need that.

Reply:
Done.